prow/github/client.go

```diff
@@ -856,13 +857,36 @@ func (c *Client) GetRepos(org string, isUser bool) ([]Repo, error) {
 
 func (c *Client) GetBranches(org, repo string) ([]Branch, error) {
 	c.log("GetBranches", org, repo)
-	var branches []Branch
-	_, err := c.request(&request{
+	var allBranches, protectedBranches []Branch
+	var err error
+	_, err = c.request(&request{
 		method:    http.MethodGet,
 		path:      fmt.Sprintf("/repos/%s/%s/branches", org, repo),
 		exitCodes: []int{200},
-	}, &branches)
-	return branches, err
+	}, &allBranches)
+	if err != nil {
+		return nil, err
+	}
+	// the list branch api does not set protected parameter on the response.
+	// Doing another call to only get protected repos
+	_, err = c.request(&request{
+		method:    http.MethodGet,
+		path:      fmt.Sprintf("/repos/%s/%s/branches?protected=true", org, repo),
```

Comment:
Please make this a boolean argument to `GetBranches()`

Move the logic to combine the two results into UpdateRepo() - https://github.com/kubernetes/test-infra/blob/master/prow/cmd/branchprotector/protect.go#L241

Reply:
that s what the GetBranches interface defines. It is just that the github client fails to implement it.

Reply:
I agree with Erick, this logic doesn't belong in the Github client. What `GetBranches` interface are you referring to?

Reply:
https://github.com/kubernetes/test-infra/blob/master/prow/cmd/branchprotector/protect.go#L146. GetBranch expects a []Branch with protected set to false or true depending on protection status. I feel that updating the interface to match the behavior of the client is backward. It complicates the branch protection code for no reason. Note also that this is bug in github, that may be resolved at some point.